CRM/Utils/System/Drupal8.php

```diff
@@ -302,17 +302,23 @@ public function url(
 
     $url = \Drupal\civicrm\CivicrmHelper::parseURL("{$path}?{$query}");
 
+    $config = CRM_Core_Config::singleton();
+    $base = $absolute ? $config->userFrameworkBaseURL : $config->useFrameworkRelativeBase;
+
     // Not all links that CiviCRM generates are Drupal routes, so we use the weaker ::fromUri method.
     try {
-      $url = \Drupal\Core\Url::fromUri("base:{$url['path']}", array(
-        'query' => $url['query'],
-        'fragment' => $fragment,
-        'absolute' => $absolute,
-      ))->toString();
+      // @todo currently there is no way to pass formatted $baseUrl as a parameter or in url,
+      //  so for now to bypass URI validation we are passing dummy string here and later replacing it with D8 base URL w/o language prefix
+      $url = str_replace('//dummy-link',
+        "{$base}{$url['path']}",
+        \Drupal\Core\Url::fromUri('//dummy-link', [
+          'query' => $url['query'],
+          'fragment' => $fragment,
+        ])->toString()
+      );
```

Comment:
Very oddly, this bit of code seems to be causing the Advance Search results to fail: run Search, then go to the second page of results. It shows a warning that the session has expired.